vic/drivers/classic/src/read_soilparam.c

```diff
@@ -618,318 +595,343 @@ read_soilparam(FILE *soilparam,
-                                   MM_PER_M;                                     // total resid_moist in mm
-                tmp_depth += temp.depth[layer];
-            }
-            b /= tmp_depth; // average b
-            bubble /= tmp_depth; // average bubble
+        /*************************************************
+           Compute soil moistures for various values of water table depth
+           Here we use the relationship (e.g., Letts et al., 2000)
+            w(z) = { ((zwt-z)/bubble)**(-1/b), z <  zwt-bubble
+                   { 1.0,                      z >= zwt-bubble
+           where
+            z      = depth below surface [cm]
+            w(z)   = relative moisture at depth z given by
+                     (moist(z) - resid_moist) / (max_moist - resid_moist)
+            zwt    = depth of water table below surface [cm]
+            bubble = bubbling pressure [cm]
+            b      = 0.5*(expt-3)
+           Note that zwt-bubble = depth of the free water surface, i.e.
+           position below which soil is completely saturated.
+
+           This assumes water in unsaturated zone above water table
+           is always in equilibrium between gravitational and matric
+           tension (e.g., Frolking et al, 2002).
+
+           So, to find the soil moisture value in a layer corresponding
+           to a given water table depth zwt, we integrate w(z) over the
+           whole layer:
+
+           w_avg = average w over whole layer = (integral of w*dz) / layer depth
+
+           Then,
+            layer moisture = w_avg * (max_moist - resid_moist) + resid_moist
+
+           Instead of the zwt defined above, will actually report free
+           water surface elevation zwt' = -(zwt-bubble).  I.e. zwt' < 0
+           below the soil surface, and marks the point of saturation
+           rather than pressure = 1 atm.
+
+           Do this for each layer individually and also for a) the top N-1 layers
+           lumped together, and b) the entire soil column lumped together.
+
+        *************************************************/
+
+        /* Individual layers */
+        tmp_depth = 0;
+        for (layer = 0; layer < options.Nlayer; layer++) {
+            b = 0.5 * (temp.expt[layer] - 3);
+            bubble = temp.bubble[layer];
+            tmp_resid_moist = temp.resid_moist[layer] * temp.depth[layer] *
+                              MM_PER_M;                                     // in mm
             zwt_prime = 0; // depth of free water surface below top of layer (not yet elevation)
             for (i = 0; i < MAX_ZWTVMOIST; i++) {
-                temp.zwtvmoist_zwt[options.Nlayer][i] = -zwt_prime; // elevation (cm) relative to soil surface
-                w_avg = (tmp_depth * CM_PER_M - zwt_prime -
+                temp.zwtvmoist_zwt[layer][i] = -tmp_depth * CM_PER_M -
+                                               zwt_prime;                         // elevation (cm) relative to soil surface
```

Comment:
OK - these are not related to either MTCLIM or removing OUTPUT_FORCE. Since they are completely unrelated, can we just make that a separate PR?